### PR TITLE
dev: add the ability to retrieve future data from data endpoint

### LINKED ISSUE #72: dev : add the ability to retrieve future entries through data endpoint
Currently we are only able to retrieve spot entries from the offchain api while populating its db with spot and future.

The purpose of this issue is to be able to query future entries also.

## TODO

- [X] be able to retrieve a price from future
- [ ] move this to a new data/future route
- [ ] add a new optionnal parameter expiry_date with default = 0 (PERP) that allow to get a price filtered by expiry date
- [ ] add a new route data/future/expiries that return a list of available expiries

### PR DESCRIPTION
**Pull request by azurwastaken**
Resolves #72 

## Added

- Added the ability to retrieve future data from data endpoint by providing entry_type parameter in the query.
- default behavior is SPOT

## Comment

There is probably refacto to do on db side to be consistent, for example we have twap_2_hours_agg and price_2_h_agg, we should have either 2_hours or 2_h for both

### CODE REVIEW

File: pragma-node/src/infra/repositories/entry_repository.rs [hunk 1]
```diff
@@ -359,89 +350,38 @@ pub async fn get_median_price(
...
-        LIMIT 1;
-    "#
-        }
-    };
+    "#,
+        get_interval_specifier(interval, false)?,
+        get_table_suffix(data_type)?
+    );
+
+    tracing::info!("get median req :{}", sql_request);
```
Comment (EvolveArt) [inline pragma-node/src/infra/repositories/entry_repository.rs:378]: Remove this

File: pragma-node/src/infra/repositories/entry_repository.rs [hunk 2]
```diff
@@ -252,89 +295,37 @@ pub async fn get_twap_price(
...
             time DESC
         LIMIT 1;
-    "#
-        }
-    };
+    "#,
+        get_interval_specifier(interval, true)?,
+        get_table_suffix(data_type)?
+    );
+    tracing::info!("get twap req :{}", sql_request);
```
Comment (EvolveArt) [inline pragma-node/src/infra/repositories/entry_repository.rs:322]: Remove

File: pragma-node/src/infra/repositories/entry_repository.rs [hunk 3]
```diff
@@ -15,12 +21,27 @@ use pragma_entities::{
...
+// Retrieve the postgres table name based on the network and data type.
+fn get_table_suffix(data_type: DataType) -> Result<&'static str, InfraError> {
+    match data_type {
+        DataType::SpotEntry => Ok(""),
+        DataType::FutureEntry => Ok("_future"),
+        _ => Err(InfraError::InternalServerError),
+    }
+}
+
+// Retrieve the postgres table name based on the network and data type.
```
Comment (EvolveArt) [inline pragma-node/src/infra/repositories/entry_repository.rs:33]: Typo

File: pragma-node/src/handlers/entries/mod.rs [hunk 4]
```diff
@@ -155,6 +177,7 @@ impl Default for GetEntryParams {
             interval: Some(Interval::default()),
             routing: Some(false),
             aggregation: Some(AggregationMode::default()),
+            entry_type: Some(EntryType::Spot),
```
Comment (EvolveArt) [inline pragma-node/src/handlers/entries/mod.rs:180]: Use default

File: pragma-node/src/handlers/entries/mod.rs [hunk 5]
```diff
@@ -131,6 +131,27 @@ pub struct Checkpoint {
     pub sender_address: String,
 }
 
+#[derive(Default, Debug, Deserialize, ToSchema, Clone, Copy)]
+pub enum EntryType {
+    #[serde(rename = "SPOT")]
```
Comment (EvolveArt) [inline pragma-node/src/handlers/entries/mod.rs:136]: Why in caps?

File: pragma-node/src/infra/repositories/entry_repository.rs (hunk 1 above)
Comment (azurwastaken) [reply pragma-node/src/infra/repositories/entry_repository.rs]: done

File: pragma-node/src/infra/repositories/entry_repository.rs (hunk 2 above)
Comment (azurwastaken) [reply pragma-node/src/infra/repositories/entry_repository.rs]: done

File: pragma-node/src/infra/repositories/entry_repository.rs (hunk 3 above)
Comment (azurwastaken) [reply pragma-node/src/infra/repositories/entry_repository.rs]: done

File: pragma-node/src/handlers/entries/mod.rs (hunk 4 above)
Comment (azurwastaken) [reply pragma-node/src/handlers/entries/mod.rs]: done

File: pragma-node/src/handlers/entries/mod.rs (hunk 5 above)
Comment (azurwastaken) [reply pragma-node/src/handlers/entries/mod.rs]: changed to lowercase, done.

**Review by EvolveArt — APPROVED** [review]

**Pull request merged**